packages/presentation/lib/home/home_page.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'bloc/home_bloc.dart';
+
+class HomePage extends StatefulWidget {
+  const HomePage({Key? key}) : super(key: key);
+
+  @override
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  final _textController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    final _bloc = context.read<HomeBloc>();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Palindrome check'),
+      ),
+      body: BlocBuilder<HomeBloc, HomeState>(
+        builder: (context, state) {
+          return Container(
+            margin: const EdgeInsets.fromLTRB(20, 50, 20, 300),
+            padding: const EdgeInsets.fromLTRB(20, 50, 20, 0),
+            alignment: Alignment.center,
+            decoration: BoxDecoration(
+              color: Colors.blue[400],
+              borderRadius: const BorderRadius.all(Radius.circular(20)),
+            ),
+            child: Column(
+              children: <Widget>[
+                TextField(
+                  controller: _textController,
+                  style: const TextStyle(color: Colors.white),
+                  decoration: const InputDecoration(
+                    enabledBorder: OutlineInputBorder(
+                      borderSide: BorderSide(color: Colors.white),
+                    ),
+                    focusedBorder: OutlineInputBorder(
+                      borderSide: BorderSide(color: Colors.white),
+                    ),
+                  ),
+                ),
+                Container(
+                  margin: const EdgeInsets.only(top: 50),
+                  child: _outputText(context, state),
+                ),
+                Container(
+                  margin: const EdgeInsets.only(top: 100),
+                  child: TextButton.icon(
+                    icon: const Icon(Icons.check, color: Colors.white),
+                    label: const Text(
+                      'Check',
+                      style: TextStyle(color: Colors.white, fontSize: 20.0),
+                    ),
+                    style: ButtonStyle(
+                      shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+                        const RoundedRectangleBorder(
+                          borderRadius: BorderRadius.all(Radius.circular(20)),
+                          side: BorderSide(color: Colors.white),
+                        ),
+                      ),
+                    ),
+                    onPressed: () {
+                      assert(_textController.text.trim().isNotEmpty);
+                      _bloc.add(CheckStringEvent(_textController.text));
```

Comment:
I think that we need to think over this logic. now the problem is that we have very strong relations between widget and bloc. in my opinion, the widget should not call the `add` method directly. the widget should call `checkPalindrome` method of bloc and bloc would trigger add method. but investigate some tutorials, maybe the way how u did it is a best practice

Reply:
Method .add just notifies bloc that something has happen and triggers appropriate handler in order to the event. I think that's a correct way according to the examples from bloclibrary.dev

Reply:
> I think that we need to think over this logic. now the problem is that we have very strong relations between widget and bloc. in my opinion, the widget should not call the `add` method directly. the widget should call `checkPalindrome` method of bloc and bloc would trigger add method. but investigate some tutorials, maybe the way how u did it is a best practice

Method .add just notifies bloc that something has happen and triggers appropriate handler in order to the event. I think that's a correct way according to the examples from bloclibrary.dev